### PR TITLE
:sparkles: add Position, AlivePlayer, Bot classes; add IPlayer interface

### PR DESCRIPTION
**Pull request by LovelyDrug**

### CODE REVIEW

File: Quoridor.Model/IPlayer.cs
Comment (VsIG-official) [inline Quoridor.Model/IPlayer.cs:6]: by the way, I don't think, that We need **Wall[]** field. Instead of this We can make **int WallsCount** and decrease this number, when placing a wall

File: Quoridor.Model/IPlayer.cs
Comment (VsIG-official) [inline Quoridor.Model/IPlayer.cs:11]: about endPosition:
I think it should be an array, 'cause We have 9 positions where a player can win

File: Quoridor.Model/Position.cs
Comment (VsIG-official) [inline Quoridor.Model/Position.cs:5]: Maybe there should be fields for x and y?

**Review by VsIG-official — COMMENTED** [review]
Wait for the @Anastasia-Tre to read this review too and then, if We'll all agree with these, You can do some rework

File: Quoridor.Model/AlivePlayer.cs
Comment (VsIG-official) [inline Quoridor.Model/AlivePlayer.cs:13]: All public fields start with [capital letter](https://docs.microsoft.com/ru-ru/dotnet/csharp/fundamentals/coding-style/coding-conventions#pascal-case)

File: Quoridor.Model/AlivePlayer.cs
Comment (VsIG-official) [inline Quoridor.Model/AlivePlayer.cs:5]: Place constructor after fields.
Also You can name parametrs as fields, but with lower letter
Like (CurrentCell = currentCell, EndPosition = endPosition) and etc.

File: Quoridor.Model/AlivePlayer.cs
Comment (VsIG-official) [inline Quoridor.Model/AlivePlayer.cs:17]: All methods starts with [capital letter](https://docs.microsoft.com/ru-ru/dotnet/csharp/fundamentals/coding-style/coding-conventions#pascal-case)

**Review by VsIG-official — COMMENTED** [review]
This review is about style, so it's up to You to do it or not

File: Quoridor.Model/IPlayer.cs
Comment (Anastasia-Tre) [reply Quoridor.Model/IPlayer.cs:6]: Agree with @VsIG-official, also we can make it so that in constructors this field has default value 10

File: Quoridor.Model/IPlayer.cs
Comment (Anastasia-Tre) [reply Quoridor.Model/IPlayer.cs:11]: +, and i think it would be better, if these fields are arrays of Cells, not Position

File: Quoridor.Model/Position.cs
Comment (Anastasia-Tre) [reply Quoridor.Model/Position.cs:5]: +

File: Quoridor.Model/IPlayer.cs
Comment (Anastasia-Tre) [inline Quoridor.Model/IPlayer.cs:5]:
```suggestion
        Cell currentCell { get; }
```

**Review by Anastasia-Tre — APPROVED** [review]

File: Quoridor.Model/AlivePlayer.cs
Comment (VsIG-official) [inline Quoridor.Model/AlivePlayer.cs:16]: Not **HasWall**, but **HasWon**

Comment (LovelyDrug) [reply Quoridor.Model/AlivePlayer.cs:16]: There is HasWon method below, I added this HasWall field for PlaceWall method. Should I remove it?

Comment (VsIG-official) [reply Quoridor.Model/AlivePlayer.cs:9]: I think so, because just [above ](https://github.com/Sierra-KPI/Quoridor/pull/10#discussion_r716197857)I suggested another way of counting walls and @Anastasia-Tre agreed with it

Comment (VsIG-official) [reply Quoridor.Model/AlivePlayer.cs:9]: But You can wait for tomorrow and then We will discuss it more

Comment (LovelyDrug) [reply Quoridor.Model/AlivePlayer.cs:9]: It's not connected with counting walls, it's just a way to check if that cell already has wall in it.

Comment (VsIG-official) [reply Quoridor.Model/AlivePlayer.cs:9]: Oh, okay, then it's My bad.
Sorry

File: Quoridor.Model/Position.cs
Comment (VsIG-official) [inline Quoridor.Model/Position.cs:5]: Return there constructor (don't delete fields) and You can merge it or wait for second review of @Anastasia-Tre

**Review by VsIG-official — APPROVED** [review]

File: Quoridor.Model/AlivePlayer.cs
Comment (LovelyDrug) [reply Quoridor.Model/AlivePlayer.cs:9]: No problem) I'm glad that you check code so thoroughly.

File: Quoridor.Model/Position.cs
Comment (VsIG-official) [reply Quoridor.Model/Position.cs:5]: But You will need to reassign Her as reviewer, if You prefer second variant

Comment (LovelyDrug) [reply Quoridor.Model/Position.cs:5]: I will wait for her

**Review by Anastasia-Tre — APPROVED** [review]

**Pull request merged**
